client/lib/analytics/track-resurrections/index.jsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from '@wordpress/element';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import getUserSetting from 'calypso/state/selectors/get-user-setting';
+
+const TrackResurrections = ( { login_time, yearInterval, isFetchingUserSettings } ) => {
+	const isResurrected = ( loginTime, previousLoginTime ) => {
+		const differenceInMilliseconds = loginTime - previousLoginTime;
+		const differenceInYears = differenceInMilliseconds / ( 1000 * 60 * 60 * 24 * 365 );
+
+		return differenceInYears > yearInterval;
+	};
+
+	useEffect( () => {
+		if ( isFetchingUserSettings ) {
+			return null;
```

Comment:
Is there a need to return `null` here? Can we just `return` void?

Reply:
Okay, two things:

1. Sorry for the confusion, I didn't mean literally `void` but rather an empty `return`:

```
if ( isFetchingUserSettings ) {
	return;
}
```

2. The link you sent applies to what we return when rendering, while the context we're talking about here is within a React `useEffect()` hook.